src/base/preferences.cpp

```diff
@@ -463,6 +463,28 @@ void Preferences::setServerDomains(const QString &str)
     setValue("Preferences/WebUI/ServerDomains", str);
 }
 
+QHostAddress Preferences::getWebUiAddress() const
+{
+    auto addr_s = value("Preferences/WebUI/Address", "*").toString();
+    if(addr_s == "*") {
+        return QHostAddress::Any;
+    }
+    QHostAddress addr;
+    if(!addr.setAddress(addr_s)) {
+        throw PreferencesParsingError(tr("Failed to parse IP Address string `%1'").arg(addr_s));
+    }
+    return addr;
```

Comment:
I would just `return  QHostAddress(addr_s);` and let `m_httpServer->listen` tell whether there are any error.